### DISCUSSION TITLE
Many thanks

### DISCUSSION THREAD

**Discussion by jfl-207** [Bodmer/TFT_eSPI#1891] [Raspberry Pi Pico]
Hi,
my firts post will not be for a question or for a problem, but for a big thumb up.
I was almost to the point of throwing away my TFT Screen
http://www.lcdwiki.com/3.5inch_Arduino_Display-Mega2560.
while it 's working fine on Arduino MEGA 2560 (16 bit parallel interface) i was absolutly unable to get it working on Pico RP2040.
All the library i've found (including your) where only able to work with the 8 bit version of this chip.
At the last moment, i've found your test version of the librairy with 16-bits support for ili9341.
Not really knowing what i was doing (i'm not a c++ guy) i've tested some modification in file "Setup103_RP2040_ILI9486_parallel.h"
And now screen is working perfectly in 16-bits.

here is my modified setup
`// This setup is for the RP2040 processor only when used with 8 bit parallel displays
// See SetupX_Template.h for all options available
#define USER_SETUP_ID 103

////////////////////////////////////////////////////////////////////////////////////////////
// Interface
////////////////////////////////////////////////////////////////////////////////////////////
//#define TFT_PARALLEL_8_BIT
#define TFT_PARALLEL_16_BIT

////////////////////////////////////////////////////////////////////////////////////////////
// Display driver type
////////////////////////////////////////////////////////////////////////////////////////////
#define ILI9486_DRIVER

////////////////////////////////////////////////////////////////////////////////////////////
// RP2040 pins used
////////////////////////////////////////////////////////////////////////////////////////////

// These pins can be moved and are controlled directly by the library software
#define TFT_RST   18  // Reset pin
#define TFT_CS    19  // Do not define if chip select control pin permanently connected to 0V
//#define TFT_RD   -1   // Do not define, read pin must be permanently connected to 3V3

// Note: All the following pins are PIO hardware configured and driven
  #define TFT_WR   16    // Write strobe pin
  #define TFT_DC   17    // Data Command control pin

  // PIO requires these to be sequentially increasing
  #define TFT_D0    0
  #define TFT_D1    1
  #define TFT_D2    2
  #define TFT_D3    3
  #define TFT_D4    4
  #define TFT_D5    5
  #define TFT_D6    6
  #define TFT_D7    7
  #define TFT_D8    8
  #define TFT_D9    9
  #define TFT_D10  10
  #define TFT_D11  11
  #define TFT_D12  12
  #define TFT_D13  13
  #define TFT_D14  14
  #define TFT_D15  15
//*/

////////////////////////////////////////////////////////////////////////////////////////////
// Fonts to be available
////////////////////////////////////////////////////////////////////////////////////////////
#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

////////////////////////////////////////////////////////////////////////////////////////////`

Once again a real BIG Thumb up from an happy user

**Comment by jfl-207** [comment]
Hi again.
Seems i was a little bit too optimist.
Grafic are fine, but i realised that colors a completly messed up.
White and black seems ok.

blue also
but Green appear yellow and Red doesn't show at all.

Any help would be nice

Regards
Jean-Francois
![IMG_20220630_010628](https://user-images.githubusercontent.com/108431943/176560822-6f736a14-11bd-4262-883a-3bf637eb3fd2.jpg)
![IMG_20220630_010633](https://user-images.githubusercontent.com/108431943/176560841-7d1b61a4-88ec-4ced-b89b-e6d5504e255a.jpg)

**Comment by Bodmer** [comment]
What is the clock frequency selected for the RP2040 in the Arduino IDE menu?

Try a lower frequency and upload the sketch again.

**Reply by jfl-207** [reply]
Hi thanks for your reply.
i was on 133, went down to 50.
still no luck except that green appear more reddish but still no red on screen.

Maybe related to RGB/BGR and or 565/666 colors ?
Way to far for my basic knowledge

regards
j-f

![IMG_20220630_153748](https://user-images.githubusercontent.com/108431943/176692079-ce1a765b-8e46-455c-8041-8d8a5d725af2.jpg)
![IMG_20220630_153753](https://user-images.githubusercontent.com/108431943/176692126-1033b026-bb25-4911-b79d-f3c64018e884.jpg)

**Comment by Bodmer** [comment]
Can you also try the latest Github master library. Save your setup somewhere safe first!
